cli/cppcheckexecutor.cpp

```diff
@@ -209,8 +209,7 @@ bool CppCheckExecutor::parseFromArgs(Settings &settings, int argc, const char* c
     }
 
     if (!pathnames.empty()) {
-        // TODO: this should be a vector or list so the order is kept
-        std::map<std::string, std::size_t> files;
+        std::list<std::pair<std::string, std::size_t>> files;
```

Comment:
I am loosely thinking maybe we should have a FileList type somewhere:

    using FileList = std::list<std::pair<std::string, std::size_t>>;

It is not a strong opinion. What do you think?

Reply:
Yeah. I thought about it after changing it but that would have just been extra work. As the goal is to merge that into `std::list<FileSettings>` I will probably adjust that at that point.